src/code42cli/cmds/auditlogs.py

```diff
@@ -0,0 +1,255 @@
+    
+    _search(
+        state.sdk,
+        format,
+        save_checkpoint,
+        begin_time=begin,
+        end_time=end,
+        event_types=event_type,
+        usernames=username,
+        user_ids=user_id,
+        user_ip_addresses=user_ip,
+        affected_user_ids=affected_user_id,
+        affected_usernames=affected_username,
+    )
+
+
+@audit_logs.command()
+@filter_options
+@checkpoint_option
+@server_options
+@send_to_format_options
+@sdk_options()
+def send_to(
+    state,
+    hostname,
+    protocol,
+    format,
+    begin,
+    end,
+    event_type,
+    username,
+    user_id,
+    user_ip,
+    affected_user_id,
+    affected_username,
+    use_checkpoint,
+):
+    """Send audit logs to the given server address."""
+    save_checkpoint = None
+    if use_checkpoint:
+        cursor = _get_audit_log_cursor_store(state.profile.name)
+        save_checkpoint = lambda ts: cursor.replace(use_checkpoint, ts)
+        if not begin:
+            begin = cursor.get(use_checkpoint)
+    
+    _send_to(
+        state.sdk,
+        hostname,
+        protocol,
+        format,
+        save_checkpoint,
+        begin_time=begin,
+        end_time=end,
+        event_types=event_type,
+        usernames=username,
+        user_ids=user_id,
+        user_ip_addresses=user_ip,
+        affected_user_ids=affected_user_id,
+        affected_usernames=affected_username,
+    )
+
+
+def _search(sdk, format, save_checkpoint, **filter_args):
+    formatter = OutputFormatter(format, AUDIT_LOGS_DEFAULT_HEADER)
+    response_gen = sdk.auditlogs.get_all(**filter_args)
+    events = _get_all_audit_log_events(response_gen)
+    event_count = len(events)
+    if not event_count:
+        click.echo("No results found.")
+        return
+    elif event_count > 10:
+        click.echo_via_pager(formatter.get_formatted_output(events))
+    else:
+        formatter.echo_formatted_list(events)
+    if save_checkpoint:
+        ts = _parse_audit_log_timestamp_string_to_timestamp(events[0]["timestamp"])
+        save_checkpoint(ts)
+
+
+def _send_to(sdk, hostname, protocol, format, save_checkpoint, **filter_args):
+    logger = get_logger_for_server(hostname, protocol, format)
+    with warn_interrupt():
+        response_gen = sdk.auditlogs.get_all(**filter_args)
+        events = _get_all_audit_log_events(response_gen)
+        if not events:
+            click.echo("No results found.")
+            return
+        for event in reversed(events):
+            logger.info(event)
+            if save_checkpoint:
+                ts = _parse_audit_log_timestamp_string_to_timestamp(event["timestamp"])
+                save_checkpoint(ts)
+
+
+def _get_all_audit_log_events(response_gen):
+    events = []
+    try:
+        for response in response_gen:
+            response_dict = json.loads(response.text)
+            if EVENT_KEY in response_dict:
+                events.extend(response_dict.get(EVENT_KEY))
+    except KeyError:
```

Comment:
It looks this will never happen because of lines 236-237.  We already checking to see if the key is in the dict

Reply:
I asked the same thing in the initial review, the exception would actually be raised from py42, but we haven't fixed that yet there, so we have to catch it here: https://github.com/code42/code42cli/pull/160#discussion_r527737186